src/AccessibilityInsights.CommonUxComponents/Controls/CustomControlOverridingAutomationPeer.cs

```diff
@@ -26,43 +28,53 @@ public class CustomControlOverridingAutomationPeer: UserControlAutomationPeer
         /// </summary>
         private string LocalizedControlType;
 
+        private AutomationControlType ControlType;
+
+        /// <summary>
+        /// Whether to hide all children automation peers
+        /// </summary>
+        private bool HideChildren;
+
         public AutomationOrientation AutomationOrientation { get; set; }
 
-        public CustomControlOverridingAutomationPeer(UserControl owner,string localizedcontrol, bool isControlElement=true, bool isContentElement=false)
+        public CustomControlOverridingAutomationPeer(
+            UserControl owner,
+            string localizedControl, 
+            bool isControlElement = true, 
+            bool isContentElement = false, 
+            bool hideChildren = false, 
+            AutomationControlType controlType = AutomationControlType.Custom)
         : base(owner)
         {
-            this.LocalizedControlType = localizedcontrol;
+            if (isContentElement && !isControlElement)
+            {
+                throw new ArgumentException("The content tree is a subset of the control tree");
```

Comment:
Should we make this a compile error rather than a runtime error? (eg, by replacinhg the 2 bool parameters with an enum of Content | ControlOnly | Decorative)

Reply:
I like the enum idea. The one (arguable) benefit I see to leaving the two bools is to keep this class as lightweight and aligned with the underlying UIAutomation classes as possible.

Reply:
The enum alternative occurred to John/me but we weren't sure if it'd be better to stick closer to the UIA surface. Now that you've also mentioned it, I'll take a closer look. We also have an existing `TreeViewMode` enum for `Raw`, `Control`, and `Content` (although `ControlOnly` might be more understandable for this one)

Reply:
That's fair enough. I think it's unlikely you'd miss this particular error in any sort of testing, so I think it's fine as a runtime check